dev/src/__tests__/updateEnrollment.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { Enrollment, User } from "../../proto/ag/ag_pb"
+import { createOvermindMock } from "overmind"
+import { config } from "../overmind"
+import { configure, render } from "enzyme"
+import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
+import { createMemoryHistory } from "history"
+import React from "react"
+import Members from "../components/Members"
+import { Route, Router } from "react-router"
+import { Provider } from "overmind-react"
+import { MockGrpcManager } from "../MockGRPCManager"
+
+
+React.useLayoutEffect = React.useEffect
+
+describe("UpdateEnrollment", () => {
+    const mockedOvermind = createOvermindMock(config, {
+        grpcMan: new MockGrpcManager()
+    })
+    it("Pending student gets accecpted", async () => {
+        await mockedOvermind.actions.getEnrollmentsByCourse({ courseID: 2, statuses: [] })
+        //This is a user with status pending
+        window.confirm = jest.fn(() => true)
+        mockedOvermind.actions.updateEnrollment({ enrollment: mockedOvermind.state.courseEnrollments[2][1], status: Enrollment.UserStatus.STUDENT })
+        await expect(mockedOvermind.state.courseEnrollments[2][1].getStatus()).toEqual(2)
+    })
+
+    it("Demote teacher to student", async () => {
+        await mockedOvermind.actions.getEnrollmentsByCourse({ courseID: 2, statuses: [] })
+        //This is a user with status teacher
+        window.confirm = jest.fn(() => true)
+        mockedOvermind.actions.updateEnrollment({ enrollment: mockedOvermind.state.courseEnrollments[2][0], status: Enrollment.UserStatus.STUDENT })
+        expect(mockedOvermind.state.courseEnrollments[2][0].getStatus()).toEqual(2)
+    })
+    it("Promote student to teacher", async () => {
+        await mockedOvermind.actions.getEnrollmentsByCourse({ courseID: 1, statuses: [] })
+        //This is a user with status student
+        window.confirm = jest.fn(() => true)
+        mockedOvermind.actions.updateEnrollment({ enrollment: mockedOvermind.state.courseEnrollments[1][0], status: Enrollment.UserStatus.TEACHER })
+        expect(mockedOvermind.state.courseEnrollments[1][0].getStatus()).toEqual(3)
+
+    })
+})
+
+configure({ adapter: new Adapter() })
+
+describe("UpdateEnrollment in webpage", () => {
+    it("If status is teacher, button should display demote", () => {
+        const user = new User().setId(1).setName("Test User").setStudentid("6583969706").setEmail("test@gmail.com")
+        const enrollment = new Enrollment().setId(2).setCourseid(1).setStatus(3).setUser(user).setSlipdaysremaining(3).setLastactivitydate("10 Mar").setTotalapproved(0)
```

Comment:
Very long line. Suggest split point:
```typescript
        const enrollment = new Enrollment().setId(2).setCourseid(1).setStatus(3).setUser(user).
            setSlipdaysremaining(3).setLastactivitydate("10 Mar").setTotalapproved(0)
```
Same applies for next test on Line 72 as well.